help_function/src/sparse_utils_2_buffer.cpp

```diff
@@ -1967,6 +1967,206 @@ void test_cusparseSpSV() {
+      handle, oneapi::mkl::transpose::nontrans,
+      oneapi::mkl::transpose::nontrans, 3, 4, 3, descrA, 4, a_c_val.d_data,
+      a_row_ptr.d_data, a_col_ind.d_data, descrB, 5, b_c_val.d_data,
+      b_row_ptr.d_data, b_col_ind.d_data, descrC, c_c_row_ptr.d_data, &c_nnz_c);
+  dpct::sparse::csrgemm_nnz(
+      handle, oneapi::mkl::transpose::nontrans,
+      oneapi::mkl::transpose::nontrans, 3, 4, 3, descrA, 4, a_z_val.d_data,
+      a_row_ptr.d_data, a_col_ind.d_data, descrB, 5, b_z_val.d_data,
+      b_row_ptr.d_data, b_col_ind.d_data, descrC, c_z_row_ptr.d_data, &c_nnz_z);
+
+  q_ct1.wait();
+
+  Data<float> c_s_val(c_nnz_s);
+  Data<double> c_d_val(c_nnz_d);
+  Data<sycl::float2> c_c_val(c_nnz_c);
+  Data<sycl::double2> c_z_val(c_nnz_z);
+  Data<int> c_s_col_ind(c_nnz_s);
+  Data<int> c_d_col_ind(c_nnz_d);
+  Data<int> c_c_col_ind(c_nnz_c);
+  Data<int> c_z_col_ind(c_nnz_z);
+
+  dpct::sparse::csrgemm(handle, oneapi::mkl::transpose::nontrans,
+                        oneapi::mkl::transpose::nontrans, 3, 4, 3, descrA,
+                        a_s_val.d_data, a_row_ptr.d_data, a_col_ind.d_data,
+                        descrB, b_s_val.d_data, b_row_ptr.d_data,
+                        b_col_ind.d_data, descrC, c_s_val.d_data,
+                        c_s_row_ptr.d_data, c_s_col_ind.d_data);
+  dpct::sparse::csrgemm(handle, oneapi::mkl::transpose::nontrans,
+                        oneapi::mkl::transpose::nontrans, 3, 4, 3, descrA,
+                        a_d_val.d_data, a_row_ptr.d_data, a_col_ind.d_data,
+                        descrB, b_d_val.d_data, b_row_ptr.d_data,
+                        b_col_ind.d_data, descrC, c_d_val.d_data,
+                        c_d_row_ptr.d_data, c_d_col_ind.d_data);
+  dpct::sparse::csrgemm(handle, oneapi::mkl::transpose::nontrans,
+                        oneapi::mkl::transpose::nontrans, 3, 4, 3, descrA,
+                        a_c_val.d_data, a_row_ptr.d_data, a_col_ind.d_data,
+                        descrB, b_c_val.d_data, b_row_ptr.d_data,
+                        b_col_ind.d_data, descrC, c_c_val.d_data,
+                        c_c_row_ptr.d_data, c_c_col_ind.d_data);
+  dpct::sparse::csrgemm(handle, oneapi::mkl::transpose::nontrans,
+                        oneapi::mkl::transpose::nontrans, 3, 4, 3, descrA,
+                        a_z_val.d_data, a_row_ptr.d_data, a_col_ind.d_data,
+                        descrB, b_z_val.d_data, b_row_ptr.d_data,
+                        b_col_ind.d_data, descrC, c_z_val.d_data,
+                        c_z_row_ptr.d_data, c_z_col_ind.d_data);
+
+  q_ct1.wait();
+
+  /*
+  DPCT1026:39: The call to cusparseDestroyMatDescr was removed because this
+  functionality is redundant in SYCL.
+  */
+  /*
+  DPCT1026:40: The call to cusparseDestroyMatDescr was removed because this
+  functionality is redundant in SYCL.
+  */
+  /*
+  DPCT1026:41: The call to cusparseDestroyMatDescr was removed because this
+  functionality is redundant in SYCL.
+  */
```

Comment:
Is this correct? I'm not sure why `cusparseDestroyMatDescr` or the equivalent `oneapi::mkl::sparse::release_matmat_descr()` APIs are redundant. Please confirm this is intentional.

Reply:
Yes, it is intentional. Because we migrate it to `std::shared_ptr<dpct::sparse::matrix_info>` so the destroy is done by `std::shared_ptr` automatically.